src/engine/cachingreaderworker.cpp

```diff
@@ -122,73 +122,89 @@ void CachingReaderWorker::run() {
 }
 
 void CachingReaderWorker::loadTrack(const TrackPointer& pTrack) {
-    ReaderStatusUpdate update;
-    update.init(TRACK_NOT_LOADED);
+    // Discard all pending read requests
+    CachingReaderChunkReadRequest request;
+    while (m_pChunkReadRequestFIFO->read(&request, 1) == 1) {
```

Comment:
is the fifo multi consumer aware? I think we need to move this into the run() method and the if (m_newTrackAvailable) { branch.

Reply:
No and it doesn't have to be. loadTrack() is private and only invoked from run() within the same thread.

Reply:
Ah yes, I misread and took it for newTrack()